M2/Macaulay2/packages/Msolve.m2

```diff
@@ -220,9 +220,11 @@ msolveRealSolutions(Ideal, Ring)       := opt -> (I0, F) -> (
     -- if precision is not specified, we want to use msolve's default precision
     prec := if precision F === infinity then msolveDefaultPrecision else precision F;
     mOut := msolve(S, K, I_*, "-p " | prec, opt);
-    -- format: [dim, [numlists, [ solution boxes ]]]
-    (d, solsp) := toSequence value readMsolveList get mOut;
+    -- format: [dim, [numlists, [ solution boxes ]]] when zero-dimensional, otherwise [1?, numgens, -1, []]
+    mSeq := toSequence value readMsolveList get mOut;
```

Comment:
So was the previous parallel assignment error occurring because `mSeq` might not have two elements?

Reply:
Yes, if the input is not zero dimensional the output has weirdly a different format entirely.